### PR TITLE
Remove some dead conditional code

### PR DESCRIPTION
**Pull request by JeremyKuhne**
CA1508 finds these, but it has many false positives from native calls and Debug asserts.

I've fixed a batch of real issues and tweaked some code to not get false positives. Won't be able to turn this one on by default, but I'll make a few more passes to find real places where there are issues. (There are over 200 hits.)

Considering making this one a suggestion, but still leaning towards no because of the false positives.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12451)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/tests/TestUtilities/PlatformDetection.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ private static bool IsLargeArrayNotSupported()
         try
         {
             byte[] tmp = new byte[int.MaxValue];
-            return tmp is null;
+            return true;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Primitives/tests/TestUtilities/PlatformDetection.cs:37]: Should this be false meaning that large array is supported?

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripMenuItemDesigner.cs [hunk 2]
```diff
@@ -440,20 +440,16 @@ internal override void CommitEdit(Type type, string text, bool commit, bool ente
...
-                if (_commitedEditorNode is not null)
-                {
-                    _commitedEditorNode.Dispose();
-                    _commitedEditorNode = null;
-                }
+                // Get rid of the templateNode.
+                _commitedTemplateNode?.CloseEditor();
+                _commitedTemplateNode = null;
+
+                _commitedEditorNode.Dispose();
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripMenuItemDesigner.cs:451]: It is, but it is checked for null already above. This is what the analyzer catches.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripKeyboardHandlingService.cs [hunk 3]
```diff
@@ -1196,111 +1181,109 @@ public void ProcessKeySelect(bool reverse)
...
         if (_shiftPressed && ShiftPrimaryItem is not null)
         {
             currentSelection = ShiftPrimaryItem;
         }
 
         currentSelection ??= SelectedDesignerControl;
 
-        ctl = currentSelection as Control;
-        if (targetSelection is null && ctl is null)
+        if (currentSelection is not Control)
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripKeyboardHandlingService.cs:1202]: `targetSelection` is always `null` at this point, but yes, I flipped control, good catch.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripKeyboardHandlingService.cs [hunk 4]
```diff
@@ -1343,142 +1326,142 @@ public void ProcessUpDown(bool down)
 
         currentSelection ??= SelectedDesignerControl;
 
-        ctl = currentSelection as Control;
+        if (currentSelection is not Control)
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripKeyboardHandlingService.cs:1329]: Same response

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**